src/bindingHandlers/graphRenderer.ts

```diff
@@ -118,44 +119,29 @@ function render(graph: LogicalGraph, elementId : string, eagle : Eagle){
         .attr("transform", rootScaleTranslation)
         .attr("class", "root")
         .attr("id", "root");
-
+        
     // add def for markers
     const defs = rootContainer.append("defs");
 
-    const black_arrowhead = defs
-        .append("marker")
-        .attr("id", "black-arrowhead")
-        .attr("viewBox", "0 0 10 10")
-        .attr("refX", "7")
-        .attr("refY", "5")
-        .attr("markerUnits", "strokeWidth")
-        .attr("markerWidth","8")
-        .attr("markerHeight", "6")
-        .attr("orient", "auto");
-
-    black_arrowhead
-        .append("path")
-        .attr("d", "M 0 0 L 10 5 L 0 10 z")
-        .attr("stroke", "none")
-        .attr("fill","black");
-
-    // add def for markers
-    const grey_arrowhead = defs
-        .append("marker")
-        .attr("id", "grey-arrowhead")
-        .attr("viewBox", "0 0 10 10")
-        .attr("refX", "7")
-        .attr("refY", "5")
-        .attr("markerUnits", "strokeWidth")
-        .attr("markerWidth","8")
-        .attr("markerHeight", "6")
-        .attr("orient", "auto");
-
-    grey_arrowhead
-        .append("path")
-        .attr("d", "M 0 0 L 10 5 L 0 10 z")
-        .attr("stroke", "none")
-        .attr("fill", LINK_COLORS['LINK_DEFAULT_COLOR']);
+    //generating defs from colors array
+    Object.keys(LINK_COLORS).forEach(function (value, i) {
+        const x = defs
```

Comment:
Maybe a better variable name here, instead of 'x', use 'arrowhead' or 'marker' or 'def'